pygmt/tests/test_grdfilter.py

```diff
@@ -0,0 +1,54 @@
+"""
+Tests for grdfilter.
+"""
+import numpy as np
+import pytest
+import xarray as xr
+from pygmt import grdfilter, grdinfo
+from pygmt.datasets import load_earth_relief
+from pygmt.exceptions import GMTInvalidInput
+from pygmt.helpers import GMTTempFile
+
+
+@pytest.fixture(scope="module", name="grid")
+def fixture_grid():
+    """
+    Load the grid data from the sample earth_relief file.
+    """
+    return load_earth_relief(registration="pixel")
+
+
+def test_grfilter_dataarray_in_dataarray_out(grid):
+    """
+    grdfilter an input DataArray, and output as DataArray.
+    """
+    result = grdfilter(grid=grid, filter="g600", distance="4")
+    # check information of the output grid
+    assert isinstance(result, xr.DataArray)
+    assert result.coords["lat"].data.min() == -89.5
+    assert result.coords["lat"].data.max() == 89.5
+    assert result.coords["lon"].data.min() == -179.5
+    assert result.coords["lon"].data.max() == 179.5
+    assert result.sizes["lat"] == 180
+    assert result.sizes["lon"] == 360
```

Comment:
Could you also check the minimum and maximum values of the output grid? The current checks can not guarantee that the output grid is correctly filtered.

Reply:
I kept getting errors when I tested those. I'm not sure if there were more decimal places than were being displayed when I printed them, which may have caused the mismatch. I'll give it another try.

Reply:
You can use some numpy testing functions (https://numpy.org/doc/stable/reference/routines.testing.html).

Reply:
The function `assert_almost_equal` worked! Thanks.